File: externals/Async-CTP/samples/CS/WCF/Stock/Quotes/WCFQuoteCalculatorService/IQuoteCalculatorService.cs
//--------------------------------------------------------------------------
// 
//  Copyright (c) Microsoft Corporation.  All rights reserved. 
// 
//  File: IQuoteCalculatorService.cs
//
//--------------------------------------------------------------------------

using System;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using System.ServiceModel;

namespace WCFTaskAsyncSample
{
    [ServiceContract]
    public interface IQuoteCalculatorService
    {
        //[OperationContract]
        //Quote GetQuote(string ticker);

        [OperationContract(AsyncPattern = true)]
        IAsyncResult BeginGetQuote(string ticker, AsyncCallback callback, object state);
        Quote EndGetQuote(IAsyncResult result);
        
        //[OperationContract]
        //ReadOnlyCollection<Quote> GetQuotes(string[] tickers);

        [OperationContract(AsyncPattern = true)]
        IAsyncResult BeginGetQuotes(string[] tickers, AsyncCallback callback, object state);
        ReadOnlyCollection<Quote> EndGetQuotes(IAsyncResult result);
    }

    [DataContract]
    public class Quote
    {
        [DataMember]
        public string Ticker { get; set; }

        [DataMember]
        public double Price { get; set; }

        [DataMember]
        public double Change { get; set; }
    }
}

File: externals/Async-CTP/samples/VB/WCF/Stock/Quotes/WCFQuoteCalculatorService/IQuoteCalculatorService.vb
'--------------------------------------------------------------------------
' 
'  Copyright (c) Microsoft Corporation.  All rights reserved. 
' 
'  File: IQuoteCalculatorService.cs
'
'--------------------------------------------------------------------------

Imports System.Collections.ObjectModel

<ServiceContract()>
Public Interface IQuoteCalculatorService

    '<OperationContract()>
    'Function GetQuote( ticker As String) As Quote

    <OperationContract(AsyncPattern:=True)>
    Function BeginGetQuote(ticker As String, callback As AsyncCallback, state As Object) As IAsyncResult
    Function EndGetQuote(result As IAsyncResult) As Quote

    '<OperationContract()>
    'Function GetQuotes( tickers As String()) As ReadOnlyCollection(Of Quote)

    <OperationContract(AsyncPattern:=True)>
    Function BeginGetQuotes(tickers As String(), callback As AsyncCallback, state As Object) As IAsyncResult
    Function EndGetQuotes(result As IAsyncResult) As ReadOnlyCollection(Of Quote)

End Interface

<DataContract()>
Public Class Quote

    <DataMember()>
    Public Property Ticker() As String

    <DataMember()>
    Public Property Price() As Double

    <DataMember()>
    Public Property Change() As Double

End Class